### PR TITLE
Adding custom image search sdk

### PR DESCRIPTION
**Pull request by q3blend**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

https://github.com/Azure/azure-rest-api-specs/pull/2999

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/CognitiveServices/dataPlane/Search/BingCustomImageSearch/BingCustomImageSearch.Tests/Microsoft.Azure.CognitiveServices.Search.CustomImageSearch.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,28 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+<Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.test.reference.props'))" />
+  <PropertyGroup>
+    <Description>Microsoft.Azure.CognitiveServices.Search.CustomImageSearch.Tests Class Library</Description>
+    <AssemblyName>Microsoft.Azure.CognitiveServices.Search.CustomImageSearch.Tests</AssemblyName>
+    <VersionPrefix>1.1.0-preview</VersionPrefix>
```
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Search/BingCustomImageSearch/BingCustomImageSearch.Tests/Microsoft.Azure.CognitiveServices.Search.CustomImageSearch.Tests.csproj]: Set the test project version to `1.0.0` this is never published

File: src/SDKs/CognitiveServices/dataPlane/Search/BingCustomImageSearch/BingCustomImageSearch/Microsoft.Azure.CognitiveServices.Search.CustomImageSearch.csproj [hunk 2]
```diff
@@ -0,0 +1,21 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <!-- Please do not move/edit code below this line -->
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
+  <!-- Please do not move/edit code above this line -->
+  
+  <PropertyGroup>
+    <PackageId>Microsoft.Azure.CognitiveServices.Search.CustomImageSearch</PackageId>
+    <Description>This client library provides access to the Microsoft Cognitive Services Custom Image Search API.</Description>
+    <VersionPrefix>1.2.0</VersionPrefix>
```
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Search/BingCustomImageSearch/BingCustomImageSearch/Microsoft.Azure.CognitiveServices.Search.CustomImageSearch.csproj]: Is this a brand new package? 
If yes, please start versioning from `1.0.0`
Is this preview?
If yes, please start versioning from `0.9.0-preview`

File: src/SDKs/_metadata/cognitiveservices_data-plane_CustomImageSearch.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+2018-05-03 09:18:53 UTC
+
+1) azure-rest-api-specs repository information
+GitHub user: Azure
+Branch:      master
+Commit:      82502085032e822e2408d3e6f5d6a6e498f01c62
+
```
Comment (dsgouda) [inline src/SDKs/_metadata/cognitiveservices_data-plane_CustomImageSearch.txt]: Please check in the generate.cmd and generate.ps1 scripts

**Review by dsgouda — CHANGES_REQUESTED** [review]
Left a few comments

File: src/SDKs/_metadata/cognitiveservices_data-plane_CustomImageSearch.txt (hunk 3 above)
Comment (q3blend) [reply src/SDKs/_metadata/cognitiveservices_data-plane_CustomImageSearch.txt]: Whats the generate.ps1? I am only aware of generate.cmd, similar to the one checked in for customsearch and other bing SDKs

File: src/SDKs/CognitiveServices/dataPlane/Search/BingCustomImageSearch/BingCustomImageSearch.Tests/Microsoft.Azure.CognitiveServices.Search.CustomImageSearch.Tests.csproj (hunk 1 above)
Comment (q3blend) [reply src/SDKs/CognitiveServices/dataPlane/Search/BingCustomImageSearch/BingCustomImageSearch.Tests/Microsoft.Azure.CognitiveServices.Search.CustomImageSearch.Tests.csproj]: Updated

File: src/SDKs/CognitiveServices/dataPlane/Search/BingCustomImageSearch/BingCustomImageSearch/Microsoft.Azure.CognitiveServices.Search.CustomImageSearch.csproj (hunk 2 above)
Comment (q3blend) [reply src/SDKs/CognitiveServices/dataPlane/Search/BingCustomImageSearch/BingCustomImageSearch/Microsoft.Azure.CognitiveServices.Search.CustomImageSearch.csproj]: Updated

File: src/SDKs/_metadata/cognitiveservices_data-plane_CustomImageSearch.txt (hunk 3 above)
Comment (q3blend) [reply src/SDKs/_metadata/cognitiveservices_data-plane_CustomImageSearch.txt]: Added generate.cmd

**Comment by dsgouda** [comment]
a sample generate.ps1 is [here](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/CognitiveServices/dataPlane/Search/BingVisualSearch/BingVisualSearch/generate.ps1)

File: src/SDKs/_metadata/cognitiveservices_data-plane_CustomImageSearch.txt (hunk 3 above)
Comment (dsgouda) [reply src/SDKs/_metadata/cognitiveservices_data-plane_CustomImageSearch.txt]: A generate.ps1 script should look something like [this](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/CognitiveServices/dataPlane/Language/Language/Language/generate.ps1)
If there is a time crunch you can do this later.

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**